code/xcvm/SPEC.md

```diff
@@ -197,24 +200,25 @@ Absolute     ::= u128
 Unit         ::= u128 Ratio
 Ratio        ::= u128 u128
 Account      ::= bytes
-Assets       ::= { AssetId : Balance }
-Transfer     ::= Account Assets | Relayer Assets
-Call         ::= Payload Bindings
+Assets       ::= { AssetId, Balance } (* optional with unique AssetId *)
+Transfer     ::= ( Account, Assets ) | ( Relayer, Assets )
+Call         ::= Payload, Bindings
 Payload      ::= bytes
-Bindings     ::= [(u16, BindingValue)]
+Bindings     ::= { u16, BindingValue } (* optional with unique u16 *)
```

Comment:
```suggestion
Bindings     ::= { u16 : BindingValue }
```

Reply:
so `:` do is key value meaning. ok. so really comment about `unique + sorted` allows not to make new `:` which is not clear exactly what. but okey.

Reply:
actually only `:` is different from some of popular EBNF formats.